### PR TITLE
feat(docker): add policy server image and compose example

### LINKED ISSUE #375: 请问后续是否计划提供Docker
您好，想问一下后续starvla是否会像rlinf那样提供docker环境呢

### PR DESCRIPTION
**Pull request by Travor278**
## Description

Adds Docker support under `deployment/docker/` for the starVLA websocket policy server.

The image is intentionally server-only: it runs `deployment/model_server/server_policy.py`, mounts checkpoints and base VLMs at runtime, and leaves benchmark simulators/eval clients in their existing host environments. This keeps the first Docker PR focused on the policy-server setup requested in #375 and avoids changing any training or evaluation code paths.

The image uses `python:3.10-slim`, installs torch/torchvision in a separate cacheable layer, and does not install a CUDA toolkit or flash-attn. The VLM inference path uses sdpa, while torch's pip wheels provide the CUDA 12.4 runtime libraries.

## Motivation

Closes #375

Setting up the policy server by hand requires several environment details that are easy to miss: matching the torch/torchvision pins, installing the repo into the container, mounting the released checkpoint layout, mounting `playground/Pretrained_models`, and keeping the websocket server alive for long evaluation sessions. This PR captures that setup in one Dockerfile plus a compose example.

## Changes

- `deployment/docker/Dockerfile.server`: policy-server image for `server_policy.py`
- `deployment/docker/docker-compose.yml`: example service with checkpoint and base-VLM mounts
- `deployment/docker/README.md`: build, run, compose, and mount instructions
- `.dockerignore`: keeps model/checkpoint/result directories out of the Docker build context

## Testing

- [x] Compose file renders with required mounts set:

```powershell
$env:MODEL_DIR='D:\dummy\model'
$env:PRETRAINED_DIR='D:\dummy\pretrained'
docker compose -f deployment/docker/docker-compose.yml config
```

- [x] Server image builds on Docker Desktop, Windows host. Local container egress could not reliably fetch the multi-GB torch/CUDA wheel set from public mirrors, so this validation build used the same torch 2.6.0 / torchvision 0.21.0 wheels from a predownloaded local wheel context. The source install, runtime entrypoint, and remaining dependency layers match the committed image path.

```powershell
docker build -f D:\Code\PR\_remote\Dockerfile.server.localtest `
  --build-context wheels=D:\Code\PR\starVLA\wheels `
  --build-arg HTTP_PROXY=http://host.docker.internal:7897 `
  --build-arg HTTPS_PROXY=http://host.docker.internal:7897 `
  -t starvla-policy-server D:\Code\PR\starVLA
```

- [x] Entrypoint help smoke test:

```powershell
docker run --rm starvla-policy-server
```

Result: `server_policy.py` prints argparse help for `--ckpt_path`, `--port`, `--use_bf16`, and `--idle_timeout`.

- [x] Runtime import smoke test:

```powershell
docker run --rm --entrypoint python starvla-policy-server -c "import torch; from deployment.model_server.policy_wrapper import PolicyServerWrapper; print('imports ok | torch', torch.__version__)"
```

Result: `imports ok | torch 2.6.0+cu124`

- [ ] Local training for N steps without errors (N/A: this PR only adds the policy-server deployment image)
- [ ] Benchmark evaluation results (N/A: this PR does not add or modify a benchmark)

| Benchmark | Metric | Result |
|-----------|--------|--------|
| N/A | N/A | N/A |

- **Checkpoint**: N/A
- **Config**: `deployment/docker/docker-compose.yml`
- **Reproduce**: see `deployment/docker/README.md`

## Type of Change

- [ ] Bug fix (non-breaking)
- [x] New feature (non-breaking)
- [ ] New framework / benchmark integration
- [ ] Breaking change
- [ ] Documentation only
- [ ] Refactor (no behavior change)

## Checklist

- [x] No unrelated changes mixed in
- [ ] New features have example config in `examples/` (N/A: deployment example lives in `deployment/docker/docker-compose.yml`)
- [x] No secrets, API keys, or large binaries committed
- [x] Files stay isolated — no unnecessary modification to shared modules (`starVLA/dataloader/`, `starVLA/training/`, `starVLA/config/`)
- [x] No modifications to shared files, or justified above (`.dockerignore` only limits Docker build context)
- [ ] If adding framework/benchmark: checkpoint uploaded to personal HF (public) (N/A)

## Screenshots / Logs (optional)

Smoke-test logs are summarized in the Testing section.

### CODE REVIEW

**Comment by Travor278** [comment]
Small scope note for review: I kept this PR to the policy-server image, compose example, and docs, matching the request in #375.

If this direction looks good, the natural follow-up would be two separate images: one training image with the CUDA devel / DeepSpeed / flash-attn stack, and one robocasa eval-client image with the robosuite source install and EGL setup. I left those out here so the first Docker review stays focused and easy to land.

**Review by JinhuiYE — APPROVED** [review]

**Comment by JinhuiYE** [comment]
Huge thanks for this great contribution! The Docker policy server setup is very welcome.
